src/libstd/sys/windows/os.rs

```diff
@@ -288,19 +287,14 @@ pub fn home_dir() -> Option<PathBuf> {
     crate::env::var_os("HOME").or_else(|| {
         crate::env::var_os("USERPROFILE")
     }).map(PathBuf::from).or_else(|| unsafe {
-        let me = c::GetCurrentProcess();
-        let mut token = ptr::null_mut();
-        if c::OpenProcessToken(me, c::TOKEN_READ, &mut token) == 0 {
-            return None
+        let mut bufArray = [0u16; 260]; // MAX_PATH
+        let buf = bufArray[..].as_mut_ptr();
+        match c::SHGetFolderPathW(ptr::null_mut(), c::CSIDL_PROFILE,
+                                  ptr::null_mut(), c::SHGFP_TYPE::SHGFP_TYPE_CURRENT as u32,
+                                  buf) {
+            0 => Some(super::os2path(&bufArray[..])),
+            _ => None
         }
-        let _handle = Handle::new(token);
-        super::fill_utf16_buf(|buf, mut sz| {
-            match c::GetUserProfileDirectoryW(token, buf, &mut sz) {
```

Comment:
Note that the definition of this function is pretty finnicky in terms of history and commitment we have, so we'll probably want to leave the current implementation anyway and have a new one for uwp

Reply:
Ok